src/main/scala/io/iohk/ethereum/jsonrpc/FilterManager.scala

```diff
@@ -222,23 +235,24 @@ class FilterManager(
 
   private def generateId(): BigInt = Math.abs(Random.nextLong())
 
-  private def resolveBlockNumber(blockParam: BlockParam, appStateStorage: AppStateStorage): BigInt = {
+  private def resolveBlockNumber(blockParam: BlockParam, bestBlockNumber: BigInt): BigInt = {
     blockParam match {
       case BlockParam.WithNumber(blockNumber) => blockNumber
       case BlockParam.Earliest => 0
-      case BlockParam.Latest => appStateStorage.getBestBlockNumber()
-      case BlockParam.Pending => appStateStorage.getBestBlockNumber() + 1
+      case BlockParam.Latest => bestBlockNumber
+      case BlockParam.Pending => bestBlockNumber
```

Comment:
For me bestBlockNumner + 1 is not exactly pending block but a special case. Let's suppose that:

- `bestBlockNumber` is 10
- you are filtering from `BlockParam.WithNumber(0)`  to `BlockParam.WithNumber(20)`
- When the time to fetch block 11 [here](https://github.com/input-output-hk/etc-client/pull/232/files#diff-b5ee4e4c8bcd462269de630eb5466a06R119), will you return `None` or `Pending` one?

I thought about tweaking the code but for me looks more confusing from a developer point of view approaching to our code. Do you think it's no?